### DISCUSSION TITLE
where can I find website with magento packages/classes/objects

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [OpenMage/magento-lts#3847] [General]
Where can I find website with magento packages/classes/objects , something like this:

http://freegento.com/doc/d8/dcb/class_mage.html

**Comment by alexh-swdev** [comment]
This one looks like a output of a PHPDoc parser, like, e.g. phpDocumentor: https://docs.phpdoc.org/3.0/guide/getting-started/generating-documentation.html#generating-documentation

So it should not be so hard to generate it by oneself? But maybe OpenMage could maintain it more-or-less automatically?

I've used such a tool eons ago, so I would'nt say I have in-depth experience with such tools, but I remember it working okayish back then.

**Reply by Flyingmana** [reply]
if we know the command, we can auto-generate it via github actions and publish it somewhere.

**Reply by alexh-swdev** [reply]
According to the docs it would be `phpdoc -d ./src -t ./docs/api` :)

But I think @kiatng is right with his comment below...

**Comment by Flyingmana** [comment]
Hi, not really an answer, more a question.
What are the use cases for having them on such a page? I always wondered what benefit they have to the sourcecode view of github.

I ask, because understanding these might help us justify putting time into it from maintenance side, and also making sure its not removed again as "no benefits"

**Comment by f1-outsourcing** [comment]
For rookies like me that are struggling with finding properties,methods,data types and what classes are available to use. Maybe best is to ask developers that start with magento

**Reply by kiatng** [reply]
When I started Magento1 many years ago, those websites generated by PHP Documenter or similar were very useful. But with modern IDEs, I don't find the need for such websites anymore. I use VS Code with extension Intelephense. The extension parses .phpstorm.meta.php., it means you get hints and F12 to navigate to the definition. The reason these features are available in the IDEs is because the OpenMage community has done a lot of works on cleaning up the doc blocks. The one drawback of IDEs is that search can be slow, but searching on github is lightning fast: https://github.com/OpenMage/magento-lts/search?q=your_search_string.

**Reply by alexh-swdev** [reply]
I think you are right... nowadays with great IDE support, those tools are almost obsolete... More helpful would be a bunch of how-to's on how accomplish common tasks with code samples. This is in part and with great explanations already done by Alana Storm (is the M1-part of her blog still online?) It helped me in the begginning. And one plugin company also provided tons of samples, but I don't remember the name anymore.

**Comment by sreichel** [comment]
Closing as work-in-progress

- https://github.com/OpenMage/magento-lts/pull/4467#issuecomment-2648431132
- https://github.com/OpenMage/magento-lts/pull/4624
